File: java/Graphs/Prim.java
Algoritmo para hallar el arbol cobertor mínimo de un grafo  no dirigido y conexo. 
Requiere de la clase Node

static int v, e; //vertices, arcos
static int MAX=100005; 
static ArrayList<Node> ady[] = new ArrayList[MAX];
static boolean marked[] = new boolean[MAX];
static int rta;
static PriorityQueue<Node> pq;

static void prim(){
	process(0); //Nodo inicial;
	int u, w;
 
	while( pq.size() > 0 ){
		u = pq.peek().adjacent;
		w = pq.peek().cost;
 
		pq.poll();
 
		if( !marked[u] ){
			rta += w;
			process(u);
		}
	}
}
 
static void process( int u ){
	marked[u] = true;
	int i, v;
 
	for( i = 0; i < ady[u].size(); i++ ){
		v = ady[u].get(i).adjacent;
        if( !marked[v] ){
			pq.add( new Node( v, ady[u].get(i).cost ) );	
		}
	}
}

File: c++/Graph/Prim.cpp
Algoritmo para hallar el arbol cobertor mínimo de un grafo  no dirigido y conexo. 
#define Node pair<int,long long> //(Vertice adyacente, peso)

int v, e; //vertices, arcos
const int MAX=100005; 
vector<Node> ady[MAX];
bool marked[MAX];
int rta;

class cmp{
public:
  bool operator()(Node n1,Node n2){
    if(n1.second>n2.second)
      return true;
    else
      return false;
  }
};

static void prim(){
    priority_queue< Node , vector<Node> , cmp > pq;
    int u, w, i, v;

    marked[0] = true;
    for( i = 0; i < ady[0].size(); i++ ){
		v = ady[0][i].first;
        if( !marked[v] )    pq.add( Node( v, ady[u][i].second ) );
	}
 
	while( !pq.empty() ){
		u = pq.top().first;
		w = pq.top().second;
 
		pq.pop();
 
		if( !marked[u] ){
			rta += w;
			marked[u] = true;
            for( i = 0; i < ady[u].size(); i++ ){
                v = ady[u][i].first;
                if( !marked[v] )    pq.add( Node( v, ady[u][i].second ) );
            }
		}
	}
}